src/mqt/bench/benchmark_generation.py

```diff
@@ -450,50 +444,97 @@ def get_benchmark(
     raise ValueError(msg)
 
 
-def timeout_watcher(
-    func: Callable[..., bool | QuantumCircuit],
-    timeout: int,
-    args: list[Any] | int | tuple[int, str] | str,
-) -> bool | QuantumCircuit:
-    """Function to handle timeouts for the benchmark generation."""
-    if sys.platform == "win32":
-        warn("Timeout is not supported on Windows.", category=RuntimeWarning, stacklevel=2)
-        return func(*args) if isinstance(args, (tuple, list)) else func(args)
-
-    class TimeoutExceptionError(Exception):  # Custom exception class
-        """Custom exception class for timeout."""
-
-    def timeout_handler(_signum: int, _frame: Any) -> None:  # noqa: ANN401
-        """Function to handle the timeout."""
-        raise TimeoutExceptionError
-
-    # Change the behavior of SIGALRM
-    signal.signal(signal.SIGALRM, timeout_handler)
-    signal.alarm(timeout)
-    try:
-        res = func(*args) if isinstance(args, (tuple, list)) else func(args)
-    except TimeoutExceptionError:
-        print(
-            "Calculation/Generation exceeded timeout limit for ",
-            func.__name__,
-            func.__module__.split(".")[-1],
-        )
-        if isinstance(args, list) and isinstance(args[0], QuantumCircuit):
-            print(args[0].name, args[1:])
-        else:
-            print(args)
-        return False
-    except Exception as e:
-        print(
-            "Exception: ",
-            e,
-            func.__name__,
-            func.__module__.split(".")[-1],
-            args,
-        )
-        return False
-    else:
-        # Reset the alarm
-        signal.alarm(0)
-
-    return res
+def get_supported_benchmarks() -> list[str]:
+    """Returns a list of all supported benchmarks."""
+    return [
+        "ae",
+        "bv",
+        "dj",
+        "grover-noancilla",
+        "grover-v-chain",
+        "ghz",
+        "graphstate",
+        "qaoa",
+        "qft",
+        "qftentangled",
+        "qnn",
+        "qpeexact",
+        "qpeinexact",
+        "qwalk-noancilla",
+        "qwalk-v-chain",
+        "randomcircuit",
+        "vqerealamprandom",
+        "vqesu2random",
+        "vqetwolocalrandom",
+        "wstate",
+        "shor",
+    ]
+
+
+def get_supported_levels() -> list[str | int]:
+    """Returns a list of all supported benchmark levels."""
+    return ["alg", "indep", "nativegates", "mapped", 0, 1, 2, 3]
+
+
+def get_default_qasm_output_path() -> str:
+    """Returns the path where all .qasm files are stored."""
+    return str(resources.files("mqt.bench") / "viewer" / "static" / "files" / "qasm_output")
+
+
+def get_zip_folder_path() -> str:
+    """Returns the path where the zip file is stored."""
+    return str(resources.files("mqt.bench") / "viewer" / "static" / "files")
```

Comment:
More of a side comment unrelated to this PR, but that I noticed because these functions were moved over from the utils module as part of this PR:
These paths no longer seem right as the viewer no longer exists and the generated files do no longer belong inside the `mqt.bench` package.
Furthermore, the name `get_default_qasm_output_path` is no longer really suitable for the future, as we will not only be exporting QASM files (QPY being the easiest extension)

Maybe they aren't needed entirely after the generation logic has been dropped.

Reply:
Great 👍🏼 then let's remove them